HpToolsLauncher/Runners/AlmTestSetsRunner.cs

```diff
@@ -154,122 +168,351 @@ public ITDConnection2 TdConnection
         }
 
 
+        //------------------------------- Connection to QC --------------------------
+
         /// <summary>
-        /// runs the tests given to the object.
+        /// Creates a connection to QC (for ALM 12.60 and 15)
         /// </summary>
-        /// <returns></returns>
-        public override TestSuiteRunResults Run()
+        private void CreateTdConnection()
         {
+            Type type = Type.GetTypeFromProgID("TDApiOle80.TDConnection");
+         
+            if (type == null)
+            {
+                ConsoleWriter.WriteLine(GetAlmNotInstalledError());
+                Environment.Exit((int)Launcher.ExitCodeEnum.Failed);
+            }
 
-            if (!Connected)
-                return null;
-            TestSuiteRunResults activeRunDesc = new TestSuiteRunResults();
-            //find all the testSets under if given some folders in our list
             try
             {
-                FindAllTestSetsUnderFolders();
+                object conn = Activator.CreateInstance(type);
+               _tdConnection = conn as ITDConnection13;
+               
             }
-            catch (Exception ex)
+            catch (FileNotFoundException ex)
             {
+                ConsoleWriter.WriteLine(GetAlmNotInstalledError());
```

Comment:
Assuming that WriteLine outputs the issue in the console output of the Jenkins job, it still would a good idea to log the issue.